### DISCUSSION TITLE
How Main App schedule run right after render app Extract Stage?

### DISCUSSION THREAD

**Discussion by younggam** [bevyengine/bevy#6368] [Q&A] [answered]
According to `update` fn in `App`
https://github.com/bevyengine/bevy/blob/a6f6a8f6e2023a53e39fd7c125077756eba40477/crates/bevy_app/src/app.rs#L145-L148
It and its sub apps use mutable reference of `World` that can't be used simultaneously in neither single or multi thread.
But, Bevy offically says that can happen after Extract Stage.
What magic used inner?

**Comment by younggam** [comment, marked as answer]
I tried to examine this own by "trace-chrome" feature
and here is the result

![제목 없음31](https://user-images.githubusercontent.com/61054554/199404573-e9a45491-ecb6-45df-8a09-0080e5478ba7.png)


As you can see, it is clear that render subapp is highly tied with main app unlike introduced below. 

https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render

I guess this news is outdated since higher version of bevy changed render polices

**Reply by mockersf** [reply]
In that blog post, there is also:

> As a quick callout, pipelined rendering doesn't actually happen in parallel yet.

It has not been done yet because... that's not the biggest thing to work on for now, and time is limited...

**Reply by younggam** [reply]
wait my eye issue
